internal/cmd/shim.go

```diff
@@ -88,16 +126,21 @@ func pluginCatalog(c *catalog.Catalog) *plugin.Catalog {
 	var schemas []*plugin.Schema
 	for _, s := range c.Schemas {
 		var enums []*plugin.Enum
+		var cts []*plugin.CompositeType
 		for _, typ := range s.Types {
-			enum, ok := typ.(*catalog.Enum)
-			if !ok {
-				continue
+			switch typ := typ.(type) {
+			case *catalog.Enum:
+				enums = append(enums, &plugin.Enum{
+					Name:    typ.Name,
+					Comment: typ.Comment,
+					Vals:    typ.Vals,
+				})
+			case *catalog.CompositeType:
```

Comment:
i assume the proto has `Enums` instead of `Types` because it it hard to represent a `Enum | CompositeType` in proto. this could also be a [`oneof`](https://developers.google.com/protocol-buffers/docs/proto3#oneof) maybe

Reply:
ah thanks, i missed the existing usage of `oneof`. i am happy to keep this as is or take a suggested refactor

Reply:
Fine to keep this for now